Src/library/base/elks/kernel/tuple.e

```diff
@@ -369,22 +443,65 @@ feature -- Element change
 			valid_index: valid_index (index)
 			valid_type_for_index: valid_type_for_index (v, index)
 		do
-			inspect eif_item_type ($Current, index)
-			when boolean_code then eif_put_boolean_item_with_object ($Current, index, $v)
-			when character_8_code then eif_put_character_8_item_with_object ($Current, index, $v)
-			when character_32_code then eif_put_character_32_item_with_object ($Current, index, $v)
-			when real_64_code then eif_put_real_64_item_with_object ($Current, index, $v)
-			when real_32_code then eif_put_real_32_item_with_object ($Current, index, $v)
-			when pointer_code then eif_put_pointer_item_with_object ($Current, index, $v)
-			when natural_8_code then eif_put_natural_8_item_with_object ($Current, index, $v)
-			when natural_16_code then eif_put_natural_16_item_with_object ($Current, index, $v)
-			when natural_32_code then eif_put_natural_32_item_with_object ($Current, index, $v)
-			when natural_64_code then eif_put_natural_64_item_with_object ($Current, index, $v)
-			when integer_8_code then eif_put_integer_8_item_with_object ($Current, index, $v)
-			when integer_16_code then eif_put_integer_16_item_with_object ($Current, index, $v)
-			when integer_32_code then eif_put_integer_32_item_with_object ($Current, index, $v)
-			when integer_64_code then eif_put_integer_64_item_with_object ($Current, index, $v)
-			when Reference_code then eif_put_reference_item_with_object ($Current, index, $v)
+			inspect item_code (index)
+			when boolean_code then
+				check valid_type_for_index: attached {BOOLEAN_REF} v as l_b then
```

Comment:
Sadly, your policy of using 'if' when the property is already expressed in the precondition only works in procedures. It does not work in functions where the result is attached.
So I was right to say that your policy is not so much about the fact that the property is already expressed in the precondition but more about not using 'check' unless the compiler complains. I'm afraid that fact that you try to reduce the use of 'check' will not prevent naive users from using it. Note that "if...else check False end" is different from "check...then...end". In the first case you always test the condition at run-time, in the second case a smart Eiffel compiler can (and if encouraged to) optimize it by verifying it only at compile-time. So it's not just a question of style.
Anyway, I will not fight to get a 'check' instead of a 'if'. There are more important things in life.